### DISCUSSION TITLE
Use fakeredis in local development

### DISCUSSION THREAD

**Discussion by cunla** [rq/rq#1809] [Ideas]
At the moment, running the tests locally requires having an active redis-server.
It would be nice to remove this need by using [fakeredis](https://github.com/cunla/fakeredis-py) - full disclaimer: I develop fakeredis.

**Comment by ccrvlh** [comment]
Hey @cunla thanks for your work on fakeredis!
On version 1.12 `rq` started using `STREAM` for the results, so it broke things with fakeredis, there was something discussion about this on #1762 .

**Comment by cunla** [comment]
fakeredis v2.7 supports streams

**Comment by ccrvlh** [comment]
Oh got it, this is fresh then! My suggestion would be to have a real server as default, with the option to fallback to fakeredis anytime - I think `pytest` can accept custom arguments on the cli, so maybe something like `pytest . --fakeredis` could work?

**Reply by cunla** [reply]
agree. It has to be somehow optional. Otherwise one won't be able to test against all versions of redis (fakeredis supports redis6/7).

**Reply by cunla** [reply]
I created an issue #1811, feel free to assign it to me.

**Comment by anton-daneyko-ultramarin** [comment]
If fakeredis is not a target platform that RQ is supposed to support, why bother trying to run the tests with it? It seems spinning up a Redis service these days is a matter of a `docker run`. So from the environment point of view it is relatively easy to set it up.
